relational/table.py

```diff
@@ -398,37 +398,84 @@ def _crunch_number(prefix):
         'num_4bd_units': OUT_4BR,
     }
 
-    def _bedroom_info_mohcd(self, rows, proj):
-        """Populates bedroom information from MOHCD.
-
-        Only pulls data from one MOHCD source, preferring Pipeline over
-        Inclusionary.  This is because this is a matter of correctness and
-        unnecessary duplication, rather than completeness.
+    def _get_mohcd_fields(self, proj, fieldmap):
+        """Extracts information from MOHCD, preferring Pipeline over
+        Inclusionary.
+
+        Returns:
+            A list of (name, value, source) tuples.  If there was not
+            at least one non-zero value, then the list will be empty,
+            regardless of whether the field existed in the MOHCD source.
         """
         out = []
         nonzero = False
         for (source, outsource) in _MOHCD_TYPES.items():
             added = False
-            for (mohcdfield, outfield) in self._MOHCD_BEDROOM_MAP.items():
+            for (mohcdfield, outfield) in fieldmap.items():
```

Comment:
Sorry I think I'm misreading the code and probably just need a quick explanation of it here, I thought we'd prefer one data source over another at the field-level (i.e. if a field is in both) but this looks like we do so at the actual source level?

Reply:
yeah, i was thinking about it, and rather than try to combine data across two different pipelines for a given project (especially since it looks like AMI buckets are not the same across all the different MOHCD datasets), just pull whatever data we get from the "best" source.

Reply:
But wouldn't this mean that if you gave it a fieldmap where pipeline had some fields and inclusionary had some fields that were not in pipeline, you'd end up not pulling the combination of the most filled out data?

Reply:
Don't want to block code submission on this and I think this code was here before this change so still going to approve, but curious if this is the way we want to do it long term (maybe we can open an issue?)

Reply:
that's right, about not selecting between both pipeline and inclusionary.

the assumption here is that pipeline and inclusionary both have their own particular snapshots of a project, and if there are fields populated in one versus the other (for identical fields) then either something meaningfully changed or we should rely on one source over the other, holistically.

i think for AMI this is absolutely the approach we need to do. between pipeline, inclusionary, and the future rental portfolio, they don't have the same number of buckets, so even identically-named fields across the datasets have different semantic meanings.

i.e. if in pipeline we see:
num_20_percent_ami_units,1
num_30_percent_ami_units,0

but in inclusionary for the same project (for some reason) we will see:
num_30_percent_ami_units,1

we should definitely not be joining the data together across the data sets, because "num_30_percent_ami_units" is working at a different level of granularity between the two datasets.

Reply:
agh, i got the specific example a little wrong, the buckets would be rolling down, not up. either way same general idea.

Reply:
er no the example is probably right. a unit affordable for 20% of ami would be by definition be affordable to 30% of ami.

Reply:
Hmm ok you're right, the more I think about it the more this makes sense (assume that each data source is internally consistent, so only use one data source without 'mixing'), i thought i had some convoluted examples in my head that wouldn't work but the more i thought about it the more I think you're right so submit away! And thanks for indulging me!